### PR TITLE
Publishers: stdout and AMS

### PR DESCRIPTION
**Pull request by enolfc**
# Summary 

This PR brings pluggable support for the output. 2 initial publishers are provided: `stdout` (the one we already had) and `ams` for pushing messages to ARGO AMS

### CODE REVIEW

**Comment by enolfc** [comment]
AMS support is broken in python3 :(
Could just use requests instead, but need to check the actual API calls needed for that.

**Comment by enolfc** [comment]
> Thanks, looks great!
> Could you please add some documentation before we merge this?

yes, will do in a subsequent commit

> And what is the status of tests?

I'm now testing with the catch-all stuff, will check with AppDB if they are still getting the same information
And some unit tests will be coming also to this repo

> Last point, should we release a version before this is merged, or should we wait for it?

I think is better to wait, as I want ONE providers to use this directly

**Comment by enolfc** [comment]
I believe now it's ready, @gwarf take a look when you can

File: README.md [hunk 1]
```diff
@@ -210,6 +210,23 @@ option of cloud-info-provider-service command (allowed values:
 `glue`, `glue21`, `cmdb`) to use a different formatter. Likewise, the default
 location of the templates can be changed using the `--template-dir` option.
 
+#### Publishers
+
+The cloud-info-provider has a pluggable system for producing its output. Two
+if this `publishers` are provided in this repo: `stdout` and `ams`:
```
Comment (gwarf) [inline README.md]:
```suggestion
of these `publishers` are provided in this repo: `stdout` and `ams`:
```

File: README.md [hunk 2]
```diff
@@ -210,6 +210,23 @@ option of cloud-info-provider-service command (allowed values:
...
+
+* `stdout`: just prints output to the standard output. This is the default
+  publisher and the one to use when the cloud-info-provider is used in a
+  BDII set up
```
Comment (gwarf) [inline README.md]:
```suggestion
  BDII set up.
```

File: setup.cfg [hunk 3]
```diff
@@ -53,6 +53,10 @@ cip.collectors =
 cip.auth_refreshers =
     oidcrefresh = cloud_info_provider.auth_refreshers.oidc_refresh:OidcRefreshToken
     oidcvorefresh = cloud_info_provider.auth_refreshers.oidc_vo_refresh:OidcVORefreshToken
+cip.publishers =
+    stdout = cloud_info_provider.publishers.stdout:StdOutPublisher
+    ams = cloud_info_provider.publishers.ams:AMSPublisher
+
```
Comment (gwarf) [inline setup.cfg]:
```suggestion
```

**Pull request merged**